spark/src/main/scala/geotrellis/spark/io/hadoop/HadoopGeoTiffRDD.scala

```diff
@@ -87,22 +92,54 @@ object HadoopGeoTiffRDD {
     * @param uriToKey function to transform input key basing on the URI information.
     * @param options  An instance of [[Options]] that contains any user defined or default settings.
     */
-  def apply[I, K, V](path: Path, uriToKey: (URI, I) => K, options: Options)(implicit sc: SparkContext, rr: RasterReader[Options, (I, V)]): RDD[(K, V)] = {
-    val conf = configuration(path, options)
-    options.maxTileSize match {
-      case Some(tileSize) =>
+  def apply[I, K, V](
+    path: Path,
+    uriToKey: (URI, I) => K,
+    options: Options
+  )(implicit sc: SparkContext, rr: RasterReader[Options, (I, V)]): RDD[(K, V)] = {
+
+    val conf = new SerializableConfiguration(configuration(path, options))
```

Comment:
Why `conf` should be wrapped here? Looks like it's only used for input formats arguments => can be used a common configuration without extra wrapper.

Reply:
That variable makes its way into a `sc.parallelize` inside of `HadoopGeoTiffInfoReader`.

Reply:
Good point, my bad.